rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -835,7 +1064,7 @@ impl SystemState {
             ingress_induction_cycles_debit,
             reserved_balance,
             reserved_balance_limit,
-            task_queue,
+            task_queue: TaskQueue::from_checkpoint(task_queue, on_low_wasm_memory_hook_status),
```

Comment:
Do we ever validate the contents of `task_queue` upon loading? If not, how can we guarantee any of the `TaskQueue` invariants?

Reply:
The check of the invariants is added when loading from the checkpoint.